### DISCUSSION TITLE
send later not working after 5/12/21 update

### DISCUSSION THREAD

**Discussion by je-github** [Extended-Thunder/send-later#331] [Q&A]
on 5/12/21, some one updated TB send later and now the current date and time no longer appear in the send later window, even if you enter it yourself it does not work, take a look.
Jamie
NJ

**Comment by jph-sendlater** [comment]
Did you intend to post a screenshot here?

**Comment by je-github** [comment]
[send later.pdf](https://github.com/Extended-Thunder/send-later/files/6489565/send.later.pdf)

**Reply by jph-sendlater** [reply]
Ah, I had another user report this to me recently. It's a bug in Thunderbird related to how it stores preferences. Basically, Send Later is unable to access any of its settings. That means nothing in send later works, but the most obvious manifestation is that it can't populate those fields with their proper defaults.

This can sometimes be caused by another extension sort of hogging access to the local storage. Do you have any other extensions installed?

**Reply by je-github** [reply]
exactly right,
 dont know of another extension, 
this just stopped working on 5/13 after the send later update.

**Reply by jph-sendlater** [reply]
It might help to see a debug log. Try opening the error console with Ctrl+Shift+J, then go to a message compose window and open and close that popup dialog a few times. Does anything show up in the error console when you do that?

**Reply by jph-sendlater** [reply]
I've been asking around with other developers, and the best suggestion I've gotten is that there's an internal component called the storage "QuotaManager" which can get corrupted. If that is indeed the problem you have three options:

The next version of Thunderbird will have the ability to self-heal that storage, so if you're patient enough to wait until later this summer when Thunderbird 95 or whatever comes out then this might fix itself.

The other option is to delete your PROFILE/storage/ directory and restart Thunderbird. This will force Thunderbird to rebuild that database, and might solve your problem right away.

Last resort is always to rebuild your profile from scratch. I would hold off on doing that for now and stick with one of the first two, but this is the only guaranteed way to solve issues like this.

**Reply by je-github** [reply]
**It might help to see a debug log. Try opening the error console with Ctrl+Shift+J, then go to a message compose window and open and close that popup dialog a few times. Does anything show up in the error console when you do that?**

IndexedDB UnknownErr: ActorsParent.cpp:567
UnknownError: The operation failed for reasons unrelated to the database itself and not covered by any other error code. ExtensionStorageIDB.jsm:831
    normalizeStorageError resource://gre/modules/ExtensionStorageIDB.jsm:831

**Reply by jph-sendlater** [reply]
Yep, that's the same error as the other user, so one of the above suggestions should apply. Let me know if you try one and what the outcome was.

**Comment by oshipper** [comment]
For me Send Later keyboard shortcut stopped working. Pressing it just sent it normally. Had to use the button on toolbar instead

Sending scheduled messages stopped as well. Just checked Drafts and full of emails that did not get sent yesterday and the ones from today as well.

I resolved this by disabling then reenabling the extension. FYI if anyone else sees similar issues

**Reply by jph-sendlater** [reply]
Thank you for reporting this, but that's a totally different issue than what this discussion thread is about. Is it fixed for you after you re-enabled the extension, or are you still having that issue?

**Comment by je-github** [comment]
any update with a TB Send/later fix?

On Sun, May 16, 2021 at 12:54 PM jph-sendlater ***@***.***>
wrote:

> Ah, I had another user report this to me recently. It's a bug in
> Thunderbird related to how it stores preferences. Basically, Send Later is
> unable to access any of its settings. That means that basically nothing in
> send later works, but the most obvious manifestation is that it can't
> populate those fields with their proper defaults.
>
> This can sometimes be caused by another extension sort of hogging access
> to the local storage. Do you have any other extensions installed?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Extended-Thunder/send-later/discussions/331#discussioncomment-744979>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUC62FHUNSSDQ3J2VMLEGWDTN72DZANCNFSM446ZXLOQ>
> .
>


-- 
*JamieEpsteinLaw.com* <http://JamieEpsteinLaw.com>

   *(856) 979-9925*

            Did you exchange

       a walk on part in a war

      for a lead role in a cage?

**Reply by jph-sendlater** [reply]
As before, this is apparently fixed in Thunderbird beta, but it will still be a while until that gets released as an ESR version (later this year some time).

Did you try the suggestions from before? Removing the `<profile>/storage` directory or rebuilding your profile from scratch?

**Reply by je-github** [reply]
5 months and numerous TB updates and Send Later not fixed

**Comment by jikamens** [comment]
Closing this discussion as Send Later is now working with current Thunderbird. Please open a new discussion (or better yet an issue, not a discussion) if you're encountering a problem with the current version.
